### ISSUE TITLE
numberAsPOSIXct(type="ncep") error

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#738]
At http://www.esrl.noaa.gov/psd/data/gridded/faq.html#3 they say:

> For example, a file could have time units of "days since 1-1-1 00:00:0.0". With those units, an encoded value of 725738. would be the value for the date 1988-01-01.

but I don't get that value; instead I get
```r
> 725738*3600*24+as.POSIXct("0001-01-01 00:00:00",tz="UTC")
[1] "1988-01-03 UTC"
```
PS. when I figure this out I guess I'll make *two* variants for ``numberAsPOSIXct()``, since it seems we have the by-hour and the by-day convention there.

@richardsc and @clayton33 -- any ideas on why my number comes out wrong by 2 days? 

PS. the present-day ``numberAsPOSIXct()`` assumes time since 1800-01-01, which is more sensible since the Julian-Gregorian shift was in the 1700s.

**Comment by richardsc** [comment]
Off the top of my head I'm not sure. But, just to get the full context, the FAQ you quoted from also says:

> Now, if you just did some quick arithmetic in your head and you came up with a different value, it's not too surprising. There are many calendar systems that have been existence at various times in history and there are various shifts that won't be accounted for by all algorithms. The translation between the times in the file and human readable times and vice versa can be greatly simplified by using the UDUNITS library from UniData. Additionally, Numerical Recipes implements the correct algorithm as well.

**Comment by dankelley** [comment]
I found an online accessor for udunits, at http://coastwatch.pfeg.noaa.gov/erddap/convert/time.html?n=725738&units=days+since+0001-01-01 and it verifies the stated test value
<img width="941" alt="screen shot 2015-09-04 at 6 43 38 pm" src="https://cloud.githubusercontent.com/assets/99469/9695094/e231260c-5334-11e5-9d7a-a7f5f0320254.png">

**Comment by dankelley** [comment]
Given my just-previous comment, I am highly leaning towards using the R code I have, and simply subtracting two days. What the hell. I assume the screwup has happened sometime in the past 2000 years, and who really cares whether R and UDUNITS disagree ... the point is that oceanographic data tend to be under 100 years old, so I don't see a terrible problem in adjusting by 2 days because of some leap-year disagreement or whatever the heck this is.

PS. I tend to trust R more than I trust the code from UNIDATA, given all the comments I've seen on blogs lately. And, seriously, what the *heck* is the point of permitting such a wide variety of time formats -- seconds since this, days since that, fortnights from the other.

**Comment by dankelley** [comment]
I've *removed* the option for ``type="ncep"`` because it was broken anyway, and this way if a user tries it, they will see that this argument is not supported. And they will see two similarly-named arguments, ``"ncep1"`` and ``"ncep2"``. These handle two of what I am guessing are going to be a multitude of NCEP conventions.  I added in a two day offset to handle the (``"ncep2"``) case of the year 0001-01-01, by subtracting 2 days. I also added some notes in the manpage. Hopefully users will be cautious on this. I also don't think the use-case is going to be very big. I ran into what I now call the ``"ncep2"`` case for NCEP reanalysis, and this is all in the past 100 years or whatever, so there shouldn't be any issues with the 2 day offset failing.

I'm closing this since it is fixed (as it is stated in the title).
